cassandra-analytics-core/src/main/java/org/apache/cassandra/spark/bulkwriter/RecordWriter.java

```diff
@@ -400,4 +403,23 @@ private StreamSession createStreamSession(TaskContext taskContext)
         LOGGER.info("[{}] Creating stream session for range={}", taskContext.partitionId(), tokenRange);
         return new StreamSession(writerContext, getStreamId(taskContext), tokenRange, failureHandler);
     }
+
+    /**
+     * Functional interface that helps with supplying {@link SSTableWriter} instances.
+     */
+    public interface SSTableWriterSupplier
```

Comment:
nit: supplier by convention takes no parameter. It is the factory here. SSTableWriterFactory wdyt?